### PR TITLE
Capture LLM metrics from genai-perf in MA

### PR DESCRIPTION
**Pull request by nv-braf**
Adding methods to capture new LLM metrics from genai-perf.

Unit testing in place as well as a small refactoring of the testing.

I've also added in the extra metrics (p95, p50, p25).

### CODE REVIEW

File: model_analyzer/perf_analyzer/perf_analyzer.py [hunk 1]
```diff
@@ -91,6 +113,28 @@ class PerfAnalyzer:
...
+        ["inter_token_latency_avg", "Inter Token Latency (ns) avg",          InterTokenLatencyAvg, "1000"],
+        ["inter_token_latency_min", "Inter Token Latency (ns) min",          InterTokenLatencyMin, "1000"],
+        ["inter_token_latency_max", "Inter Token Latency (ns) max",          InterTokenLatencyMax, "1000"],
+        ["inter_token_latency_p99", "Inter Token Latency (ns) p99",          InterTokenLatencyP99, "1000"],
+        ["inter_token_latency_p95", "Inter Token Latency (ns) p95",          InterTokenLatencyP95, "1000"],
+        ["inter_token_latency_p90", "Inter Token Latency (ns) p90",          InterTokenLatencyP90, "1000"],
+        ["inter_token_latency_p75", "Inter Token Latency (ns) p75",          InterTokenLatencyP75, "1000"],
+        ["inter_token_latency_p50", "Inter Token Latency (ns) p50",          InterTokenLatencyP50, "1000"],
+        ["inter_token_latency_p25", "Inter Token Latency (ns) p25",          InterTokenLatencyP25, "1000"],
+        ["output_token_throughput", "Output Token Throughput (per sec) avg", OutputTokenThroughput, "1"]
```
Comment (tgerdesnv) [inline model_analyzer/perf_analyzer/perf_analyzer.py:136]: Is the 'avg' here for reading the csv? Or is it an output header?

File: model_analyzer/record/types/inter_token_latency_p95.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            max, min or average etc.
+
+        Returns
+        -------
+        str
+            The full name of the
+            metric.
+        """
+
+        return "p90 Inter Token Latency (ms)"
```
Comment (tgerdesnv) [inline model_analyzer/record/types/inter_token_latency_p95.py:60]: wrong header?

File: model_analyzer/record/types/time_to_first_token_p25.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            max, min or average etc.
+
+        Returns
+        -------
+        str
+            The full name of the
+            metric.
+        """
+
+        return "p99 Time To First Token (ms)"
```
Comment (tgerdesnv) [inline model_analyzer/record/types/time_to_first_token_p25.py:60]: wrong

File: model_analyzer/record/types/time_to_first_token_p50.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            max, min or average etc.
+
+        Returns
+        -------
+        str
+            The full name of the
+            metric.
+        """
+
+        return "p99 Time To First Token (ms)"
```
Comment (tgerdesnv) [inline model_analyzer/record/types/time_to_first_token_p50.py:60]: wrong

File: model_analyzer/record/types/time_to_first_token_p95.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            max, min or average etc.
+
+        Returns
+        -------
+        str
+            The full name of the
+            metric.
+        """
+
+        return "p99 Time To First Token (ms)"
```
Comment (tgerdesnv) [inline model_analyzer/record/types/time_to_first_token_p95.py:60]: ^^

File: model_analyzer/perf_analyzer/perf_analyzer.py (hunk 1 above)
Comment (nv-braf) [reply model_analyzer/perf_analyzer/perf_analyzer.py:136]: It's just to match reading the CSV.

Comment (tgerdesnv) [reply model_analyzer/perf_analyzer/perf_analyzer.py:136]: ah ok, so you aren't using the 2nd table header. No worries. It is ugly, but hopefully soon we will go to something much cleaner.

Comment (nv-braf) [reply model_analyzer/perf_analyzer/perf_analyzer.py:136]: Correct. DictReader assumes the first row is the header, and it ignores the blank row. It's definitely ugly, but it works (and is less ugly than having to hand read in the file and figure this all out myself).

File: model_analyzer/record/types/inter_token_latency_p25.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
```
Comment (tgerdesnv) [inline model_analyzer/record/types/inter_token_latency_p25.py:1]: is this needed?

Comment (nv-braf) [reply model_analyzer/record/types/inter_token_latency_p25.py:1]: I don't know. They are the first line in all of the `.py` files

**Review by tgerdesnv — APPROVED** [review]

**Pull request merged**